framework/include/base/MooseApp.h

```diff
@@ -708,11 +708,13 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
    */
   bool addRelationshipManager(std::shared_ptr<RelationshipManager> relationship_manager);
 
+private:
   /**
    * Purge this relationship manager from meshes and DofMaps and finally from us
    */
   void removeRelationshipManager(std::shared_ptr<RelationshipManager> relationship_manager);
```

Comment:
It would be a big plus if you could add a comments about what siuation we need to remove RMs.

In my head, I alway think adding and adding. I do not have use cases where I need to remove RMs.